src/py21cmfast/src/SpinTemperatureBox.c

```diff
@@ -1123,36 +1403,49 @@ LOG_SUPER_DEBUG("looping over box...");
 
 
                 ave_fcoll /= (pow(10.,10.)*(double)HII_TOT_NUM_PIXELS);
+                ave_fcoll_MINI /= (pow(10.,10.)*(double)HII_TOT_NUM_PIXELS);
 
                 if(ave_fcoll!=0.) {
                     ave_fcoll_inv = 1./ave_fcoll;
                 }
 
+                if(ave_fcoll_MINI!=0.) {
+                    ave_fcoll_inv_MINI = 1./ave_fcoll_MINI;
+                }
+
                 dfcoll_dz_val = (ave_fcoll_inv/pow(10.,10.))*ST_over_PS[R_ct]*SFR_timescale_factor[R_ct]/astro_params->t_STAR;
 
                 dstarlya_dt_prefactor[R_ct] *= dfcoll_dz_val;
 
+                if(flag_options->USE_MINI_HALOS){
+                    dfcoll_dz_val_MINI = (ave_fcoll_inv_MINI/pow(10.,10.))*ST_over_PS_MINI[R_ct]*SFR_timescale_factor[R_ct]/astro_params->t_STAR;
+                    dstarlya_dt_prefactor_MINI[R_ct] *= dfcoll_dz_val_MINI;
+                    dstarlyLW_dt_prefactor[R_ct] *= dfcoll_dz_val;
+                    dstarlyLW_dt_prefactor_MINI[R_ct] *= dfcoll_dz_val_MINI;
+                }
+
                 for (box_ct=HII_TOT_NUM_PIXELS; box_ct--;){
 
                     // I've added the addition of zero just in case. It should be zero anyway, but just in case there is some weird
                     // numerical thing
                     if(ave_fcoll!=0.) {
                         dxheat_dt_box[box_ct] += (dfcoll_dz_val*(double)del_fcoll_Rct[box_ct]*( (freq_int_heat_tbl_diff[m_xHII_low_box[box_ct]][R_ct])*inverse_val_box[box_ct] + freq_int_heat_tbl[m_xHII_low_box[box_ct]][R_ct] ));
                         dxion_source_dt_box[box_ct] += (dfcoll_dz_val*(double)del_fcoll_Rct[box_ct]*( (freq_int_ion_tbl_diff[m_xHII_low_box[box_ct]][R_ct])*inverse_val_box[box_ct] + freq_int_ion_tbl[m_xHII_low_box[box_ct]][R_ct] ));
-                    }
-                    else {
-                        dxheat_dt_box[box_ct] += 0.;
```

Comment:
You have deleted this else statement. Is there a reason why you have done this? This was here to ensure that zero was added if there was some weird numerical issue